### PR TITLE
Add travis build support

### PR DESCRIPTION
**Pull request by parabolic**
- travis tests on every commit
- travis will push a release when tagged and on master branch
- add a darwin build option in the GNUmakefile

### CODE REVIEW

File: GNUmakefile [hunk 1]
```diff
@@ -8,7 +8,11 @@ sanitycheck:
 	$(MAKE) depensure
 	$(MAKE) fmtcheck
 
-build: sanitycheck
+
+build-darwin:
+	GOOS=darwin GOARCH=amd64 go install
+
+build:
```
Comment (jansiwy) [inline GNUmakefile]: could we call it `build-linux` for symmetry?

File: README.md [hunk 2]
```diff
@@ -66,3 +66,14 @@ $ make test
 We cannot have the acceptance tests until rollbar changes/improves their api.
 The reason is because creating an invitation doesn't yield an userid.
 A user id is created when a user accepts their invitation.
+
+Github Releases
+---------------------------
+In order to push a release to github the feature branch has to merged into master and then a tag needs to be created the same as the provider version and pushed as well.
```
Comment (jansiwy) [inline README.md]: github => GitHub

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+before_deploy:
+  - GIT_TAG=$(git describe --tags)
+  - make build
+  - mv ${LINUX_BINARY_PATH} ${LINUX_BINARY_PATH}_v${GIT_TAG}
+  - zip -j ${LINUX_BINARY_PATH}_v${GIT_TAG}_linux_amd64.zip ${LINUX_BINARY_PATH}_v${GIT_TAG}
+  - make build-darwin
+  - mv ${DARWIN_BINARY_PATH} ${DARWIN_BINARY_PATH}_v${GIT_TAG}
+  - zip -j ${DARWIN_BINARY_PATH}_v${GIT_TAG}_darwin_amd64.zip ${DARWIN_BINARY_PATH}_v${GIT_TAG}
+
+# Deploy only when there's a tag on master.
```
Comment (jansiwy) [inline .travis.yml]: what is a "tag on master"? if mean if the tag has the identical commit SHA is the current master?

what about: if the tag names looks like version? something like `/v(\d+)\.(\d+)\.(\d+)/` instead?

**Review by jansiwy — APPROVED** [review]
beyond that: +B

File: .travis.yml [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+before_deploy:
+  - GIT_TAG=$(git describe --tags)
+  - make build
+  - mv ${LINUX_BINARY_PATH} ${LINUX_BINARY_PATH}_${GIT_TAG}
+  - zip -j ${LINUX_BINARY_PATH}_${GIT_TAG}_linux_amd64.zip ${LINUX_BINARY_PATH}_${GIT_TAG}
+  - make build-darwin
+  - mv ${DARWIN_BINARY_PATH} ${DARWIN_BINARY_PATH}_${GIT_TAG}
+  - zip -j ${DARWIN_BINARY_PATH}_${GIT_TAG}_darwin_amd64.zip ${DARWIN_BINARY_PATH}_${GIT_TAG}
+
+# Deploy only whrn a tag is created and pushed to the master branch.
```
Comment (jtsaito) [inline .travis.yml]: `whrn` -> `when` :)

**Review by jtsaito — APPROVED** [review]

**Pull request merged**